packages/horizon/scripts/e2e/pre-upgrade.ts

```diff
@@ -0,0 +1,197 @@
+import { mergeABIs } from 'hardhat-graph-protocol/sdk'
+import StakingExtensionABI from '@graphprotocol/contracts/build/abis/StakingExtension.json'
+
+import { createBanner } from '../../utils/banners'
+import { delegators } from './fixtures/delegators'
+import { indexers } from './fixtures/indexers'
+
+// The account on Arbitrum Sepolia that has GRT tokens
+const GRT_HOLDER_ADDRESS = process.env.GRT_HOLDER_ADDRESS || '0xadE6B8EB69a49B56929C1d4F4b428d791861dB6f'
+
+// Load ABIs
+const combinedStakingABI = mergeABIs(L2StakingABI, StakingExtensionABI)
+const graphTokenABI = L2GraphTokenABI
+
+// Generate allocation proof with the indexer's address and the allocation id, signed by the allocation private key
+function generateAllocationProof(indexerAddress: string, allocationPrivateKey: string) {
+  const wallet = new ethers.Wallet(allocationPrivateKey)
+  const messageHash = ethers.solidityPackedKeccak256(
+    ['address', 'address'],
+    [indexerAddress, wallet.address],
+  )
+  const messageHashBytes = ethers.getBytes(messageHash)
+  return wallet.signMessage(messageHashBytes)
+}
+
+const randomHexBytes = (n = 32): string => ethers.hexlify(ethers.randomBytes(n))
+
+async function main() {
+  console.log(createBanner('PRE-HORIZON UPGRADE SETUP'))
+
+  console.log('\n--- STEP 0: Setup ---')
+
+  // Verify that hardhat network accounts are set to remote, otherwise we won't be able to impersonate them
+  if (hre.network.config.accounts !== 'remote') {
+    throw new Error('Hardhat network accounts must be set to remote')
+  }
+
+  // Load contract addresses from addresses.json
+  const addressesJson = require('@graphprotocol/contracts/addresses.json')
+  const arbSepoliaAddresses = addressesJson['421614']
+
+  // Get contract addresses
+  const stakingAddress = arbSepoliaAddresses.L2Staking.address
+  const graphTokenAddress = arbSepoliaAddresses.L2GraphToken.address
+
+  console.log(`Using Staking contract at: ${stakingAddress}`)
+  console.log(`Using GraphToken contract at: ${graphTokenAddress}`)
+
+  // Create contract instances
+  const provider = new HardhatEthersProvider(hre.network.provider, hre.network.name)
+  const GraphToken = new Contract(graphTokenAddress, graphTokenABI, provider) as unknown as IGraphToken
+  const Staking = new Contract(stakingAddress, combinedStakingABI, provider) as unknown as IStaking
+
+  // The account on Arbitrum Sepolia that has GRT tokens
+  const assetHolderBalance = BigInt((await GraphToken.balanceOf(GRT_HOLDER_ADDRESS)).toString())
+  console.log(`Asset holder balance: ${assetHolderBalance}`)
+
+  // Convert BigNumber to bigint for comparison
+  if (assetHolderBalance < ethers.parseEther('20000000')) {
+    throw new Error('Asset holder balance is less than 20M tokens')
+  }
+
+  // Impersonate the account
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const grtHolder = await ethers.getImpersonatedSigner(GRT_HOLDER_ADDRESS) as any
```

Comment:
How about instead of relying on a known account with big chunk of GRT we use the governor to mint some?